core/silkworm/types/block.cpp

```diff
@@ -37,6 +37,9 @@ ethash::hash256 BlockHeader::boundary() const {
 
     if (difficulty > 1u) {
         auto result{intx::bswap(dividend / difficulty)};
+        if ((difficulty & (difficulty - 1)) == 0) {
+            result += 1;
+        }
```

Comment:
This is not correct as you're adding 1 **after** the byte swapping while instead you should add 1 to the quotient.
```suggestion
        unsigned int add_one{(difficulty & (difficulty - 1)) == 0 ? 1u : 0};
        auto result{intx::bswap((dividend / difficulty) + add_one)};
```

Also I realized `if (difficulty > 1u)` should be instead `if (difficulty >= 1u)`

Please also add tests

Reply:
ops, i misunderstood what you said in the issue, test are coming shortly